frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { Form, FormSelect, FormSelectOption, FormSelectProps, Radio } from '@patternfly/react-core';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import { StorageClassResourceKind } from '@console/internal/module/k8s';
+import './backing-storage.scss';
+import { ExternalProvider } from 'packages/ceph-storage-plugin/src/odf-external-providers/types';
+import { ODF_EXTERNAL_PROVIDERS } from '../../../../odf-external-providers/external-providers';
+import { StorageSystemKind } from '../../../../types';
+import { filterSCWithoutNoProv } from '../../../../utils/install';
+import { WizardReducer, WizardState, WizardDispatch } from '../../reducer';
+import {
+  BackingStorageType,
+  RHCS,
+  StorageClusterIdentifier,
+} from '../../../../constants/create-storage-system';
+import { ErrorHandler } from '../../error-handler';
+
+const ExternalSystemSelection: React.FC<ExternalSystemSelectionProps> = ({
+  dispatch,
+  providersList,
+  selectedProvider,
+  hasOCS,
+}) => {
+  const { t } = useTranslation();
+  const handleSelection: FormSelectProps['onChange'] = (value: string) =>
+    dispatch({
+      type: 'backingStorage/setExternalProvider',
+      payload: value,
+    });
+
+  return (
+    <FormSelect
+      aria-label={t('ceph-storage-plugin~Select external system from list')}
+      value={selectedProvider}
+      className="odf-backing-storage__selection--width"
+      onChange={handleSelection}
+    >
+      {hasOCS && <FormSelectOption key={RHCS} value={RHCS} label="Red Hat Ceph Storage" />}
+      {providersList.map((p) => (
+        <FormSelectOption key={p.id} value={p.id} label={p.label} />
+      ))}
+    </FormSelect>
+  );
+};
+
+type ExternalSystemSelectionProps = {
+  dispatch: React.Dispatch<React.ReducerAction<WizardReducer>>;
+  selectedProvider: WizardState['backingStorage']['externalProvider'];
+  providersList: ExternalProvider[];
+  hasOCS: boolean;
+};
+
+const StorageClassSelection: React.FC<StorageClassSelectionProps> = ({ dispatch, selected }) => {
+  const onStorageClassSelect = (sc: StorageClassResourceKind) =>
+    dispatch({
+      type: 'wizard/setStorageClass',
+      payload: { name: sc?.metadata?.name, provisioner: sc?.provisioner },
+    });
+  return (
+    <div className="odf-backing-storage__selection--width">
+      <StorageClassDropdown
+        noSelection
```

Comment:
nit: move noSelection to the end of parameter list

Reply:
Why do we ?

Reply:
It looks a bit cleaner to me. It's fine it can be ignored.